### PR TITLE
Refactoring how Phing handles boolean strings

### PR DESCRIPTION
**Pull request by jawira**
Hi. As you know Phing handles some strings as boolean. According to documentation:

> Note: There are special values that are not mapped to strings, but to boolean values instead. The values true, false, yes, no, on and off are translated to true/false boolean values.

Source: https://www.phing.info/guide/chunkhtml/ch05s05.html

However, when I checked the code I found that booleans are converted in two places:

1. `\Phing\Project::toBoolean`
2. `\Phing\Util\StringHelper::booleanValue`

So I did a refactoring and I put the code in a single place: `StringHelper::booleanValue`. Sadly I noticed that both methods don't behave exactly the same way.

- `Project::toBoolean` behaves almost as stated in documentation: true, 'true', '1', 'yes', 'on' are considered true, and everything else is false. I kept this behaviour.
- `StringHelper::booleanValue` does the same as `Project::toBoolean`, but also considers `t` and `f` as bool (which, IMHO makes no sense).

Please see the tests I wrote to see how these methods behave.

I didn't remove `Project::toBoolean` because I don't know if any task (having a independent repo) is using it. Instead this method calls and returns the value from `StringHelper::booleanValue`. 

I also seized the opportunity and cleaned the code in `\Phing\Util\StringHelper::isBoolean

### CODE REVIEW

File: src/Phing/Project.php [hunk 1]
```diff
@@ -863,13 +864,7 @@ public function resolveFile(string $fileName, File $rootDir = null): File
      */
     public static function toBoolean($s)
```
Comment (siad007) [inline src/Phing/Project.php:865]: @jawira feels good but after short investigation I found some locations where the `Project::toBoolean` is also used.
See:
```
Method
    toBoolean
Found usages  (9 usages found)
    Method call  (9 usages found)
        phing  (9 usages found)
            src  (9 usages found)
                Phing  (9 usages found)
                    Filter  (3 usages found)
                        LineContains.php  (1 usage found)
                            LineContains  (1 usage found)
                                initialize  (1 usage found)
                                    265 $this->setNegate(Project::toBoolean($param->getValue()));
                        LineContainsRegexp.php  (2 usages found)
                            LineContainsRegexp  (2 usages found)
                                initialize  (2 usages found)
                                    270 $this->setNegate(Project::toBoolean($params[$i]->getValue()));
                                    272 $this->setCaseSensitive(Project::toBoolean($params[$i]->getValue()));
                    Task  (2 usages found)
                        System  (2 usages found)
                            Condition  (2 usages found)
                                IsPropertyFalseCondition.php  (1 usage found)
                                    IsPropertyFalseCondition  (1 usage found)
                                        evaluate  (1 usage found)
                                            60 return null === $value || !Project::toBoolean($value);
                                IsPropertyTrueCondition.php  (1 usage found)
                                    IsPropertyTrueCondition  (1 usage found)
                                        evaluate  (1 usage found)
                                            57 return null !== $value && Project::toBoolean($value);
                    Type  (4 usages found)
                        Selector  (4 usages found)
                            ContainsRegexpSelector.php  (2 usages found)
                                ContainsRegexpSelector  (2 usages found)
                                    setParameters  (2 usages found)
                                        136 $this->setCasesensitive(Project::toBoolean($parameters[$i]->getValue()));
                                        141 $this->setMultiLine(Project::toBoolean($parameters[$i]->getValue()));
                            FilenameSelector.php  (2 usages found)
                                FilenameSelector  (2 usages found)
                                    setParameters  (2 usages found)
                                        149 $this->setCasesensitive(Project::toBoolean($parameters[$i]->getValue()));
                                        154 $this->setNegate(Project::toBoolean($parameters[$i]->getValue()));

```
Please change to the StringHelper on those locations as well and annotate the `Project::toBoolean` as deprecated.

**Review by siad007 — CHANGES_REQUESTED** [review]

File: src/Phing/Project.php (hunk 1 above)
Comment (jawira) [reply src/Phing/Project.php:866]: Done!

**Comment by siad007** [comment]
@jawira I just resolved the merge conflict but there was added an additional `trim()` by excident - would you mind to remove it please? I will then merge this PR.

**Comment by jawira** [comment]
@siad007 I removed the `trim` :)

**Review by siad007 — APPROVED** [review]

**Comment by siad007** [comment]
thx @jawira

**Pull request merged**